src/components/smart_objects/include/smart_objects/enum_schema_item.h

```diff
@@ -325,6 +333,40 @@ Errors::eType TEnumSchemaItem<EnumType>::validate(
   return Errors::OK;
 }
 
+template <typename EnumType>
+const ElementSignature TEnumSchemaItem<EnumType>::getSignature(
+    const std::vector<ElementSignature>& signatures,
+    const utils::SemanticVersion& MessageVersion) {
+  for (uint i = 0; i < signatures.size(); i++) {
+    ElementSignature signature = signatures[i];
+    // Check if signature matches message version
+    if (signature.mSince != boost::none) {
+      if (MessageVersion < signature.mSince.get()) {
+        // Msg version predates 'since' field, check next entry
+        continue;
+      }
+
+      if (signature.mUntil != boost::none &&
+          (MessageVersion >= signature.mUntil.get())) {
+        continue;  // Msg version newer than `until` field
+      }
+
+      return signature;
+    }
+
+    if (signature.mUntil != boost::none &&
+        (MessageVersion >= signature.mUntil.get())) {
+      continue;  // Msg version newer than `until` field, check next entry
+    }
+
+    return signature;
```

Comment:
Overcomplicated, could be reduced to:
```
    ElementSignature signature = signatures[i];
    // Check if signature matches message version
    if (signature.mUntil != boost::none &&
        (MessageVersion < signature.mSince.get()) {
      // Msg version predates 'since' field, check next entry
      continue;
    }
    if (signature.mUntil != boost::none &&
        (MessageVersion >= signature.mUntil.get())) {
      continue;  // Msg version newer than `until` field, check next entry
    }
    return signature;
```